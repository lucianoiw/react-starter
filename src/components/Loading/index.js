import React from 'react';
import PropTypes from 'prop-types';

import nullablePropTypes from '@Utils/nullablePropTypes';

const Loading = ({
  error,
  timedOut,
}) => {
  if (error || timedOut) {
    return (
      <>
        Ocorreu um erro!
      </>
    );
  }

  return (
    <>
      Carregando...
    </>
  );
};

Loading.propTypes = {
  timedOut: PropTypes.bool.isRequired,
  error: nullablePropTypes(PropTypes.object).isRequired,
};

export default Loading;
