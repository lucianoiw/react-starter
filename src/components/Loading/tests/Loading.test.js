import React from 'react';
import { shallow } from 'enzyme';
import Loading from '../index';

it('snapshot loading', () => {
  const props = {
    isLoading: true,
    error: null,
    timedOut: false,
  };

  const rendered = shallow(<Loading {...props} />);
  expect(rendered).toMatchSnapshot();
});

it('snapshot error', () => {
  const props = {
    isLoading: false,
    error: {
      error: true,
    },
    timedOut: false,
  };

  const rendered = shallow(<Loading {...props} />);
  expect(rendered).toMatchSnapshot();
});

it('snapshot timeout', () => {
  const props = {
    isLoading: false,
    error: null,
    timedOut: true,
  };

  const rendered = shallow(<Loading {...props} />);
  expect(rendered).toMatchSnapshot();
});
