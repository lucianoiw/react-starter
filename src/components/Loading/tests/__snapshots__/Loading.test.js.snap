// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot error 1`] = `
<Fragment>
  Ocorreu um erro!
</Fragment>
`;

exports[`snapshot loading 1`] = `
<Fragment>
  Carregando...
</Fragment>
`;

exports[`snapshot timeout 1`] = `
<Fragment>
  Ocorreu um erro!
</Fragment>
`;
